content/en/blog/_posts/2020-06-29-overview-working-with-terraform-and-kubernetes.md

```diff
@@ -0,0 +1,59 @@
+
+This provider uses dynamic resource types and server-side-applies to support all Kubernetes resources. I personally think this provider has the potential to be a game changer. Even if managing [Kubernetes resources in HCL](https://github.com/hashicorp/terraform-provider-kubernetes-alpha#moving-from-yaml-to-hcl) may still not be for everyone. Maybe the Kustomize provider below may help with that.
+
+The only downside really is, that it's explicitly discouraged to use it for anything but testing. But the more people test it, the sooner it should be ready for prime time. So I encourage everyone to give it a try.
+
+## Terraform Provider Kustomize
+
+Last, the [Terraform Kustomize provider](https://github.com/kbst/terraform-provider-kustomize). Kustomize provides a way to do customizations of Kubernetes resources using inheritance, instead of templating. It is designed to output the result to `stdout`, from where Kubectl can be used to apply the changes. This means Kubectl edge cases like no purging or changes to immutable attributes still make full automation difficult.
+
+Kustomize is a popular way to handle customizations. But I was looking for a more reliable way to automate applying changes. Since this is exactly what Terraform is great at the Kustomize provider was born.
+
+Not going into too much detail here, but from Terraform's perspective, this provider treats every Kubernetes resource as a JSON string. This way it can handle any Kubernetes resource resulting from the Kustomize build. But it has the big disadvantage that Kubernetes resources can not easily be integrated with other Terraform resources. Remember the load balancer example from above.
+
+Under the hood, similarly to the new Kubernetes alpha provider, the Kustomize provider also uses the dynamic Kubernetes client and server-side-applies. Going forward, I plan to deprecate this part of the Kustomize provider that overlaps with the new Kubernetes provider and only keep the Kustomize integration.
+
+## Conclusion
+
+For teams that are already invested into Terraform, or teams that are looking for ways to replace Kubectl in automation, Terraform's plan/apply life-cycle has always been a promising option to automate changes to Kubernetes resources. However, the limitations of the official Kubernetes provider resulted in this not seeing significant adoption.
+
+The new alpha provider removes the limitations and has the potential to make Terraform a prime option to automate changes to Kubernetes resources.
+
+Teams that have already adopted Kustomize, may find integrating Kustomize and Terraform using the Kustomize provider beneficial over Kubectl because it avoids common edge cases. Even if in this set up, Terraform can only easily be used to plan and apply the changes, not to adapt the Kubernetes resources. In the future, this issue may be resolved by combining the Kustomize provider with the new Kubernetes provider.
```

Comment:
```suggestion
Teams that have already adopted Kustomize, may find integrating Kustomize and Terraform using the Kustomize provider beneficial over `kubectl` because it avoids common edge cases. Even if in this set up, Terraform can only easily be used to plan and apply the changes, not to adapt the Kubernetes resources. In the future, this issue may be resolved by combining the Kustomize provider with the new Kubernetes provider.
```